packages/eslint-plugin/package.json

```diff
@@ -27,6 +28,18 @@
       "default": "./dist/configs/eslint-recommended-raw.js"
     }
   },
+  "typesVersions": {
+    "*": {
+      "use-at-your-own-risk/rules": [
+        "./rules.d.ts"
+      ],
+      "use-at-your-own-risk/eslint-recommended-raw": [
+        "./eslint-recommended-raw.d.ts"
+      ]
+    }
+  },
```

Comment:
Why do we need a duplicate definition here? These are already defined via the `exports` key.

Reply:
`Node10` module resolution would fail without them

![tseslint-without-typesVersions](https://github.com/user-attachments/assets/1362ce01-5955-4d98-b487-43b8b6cf55b4)

Reply:
> Node10 module resolution would fail without them

We do not support Node10.
Node10 as a version of node is LONG deprecated.
Node10 as a module resolution is also bad because it means your module resolution doesn't match reality.

Reply:
That's understandable. However, from my perspective, the benefits of supporting `Node10` module resolution still outweigh the cons for now. While `Node10` itself is deprecated, there are still a significant number of codebases that rely on it. A quick search on GitHub reveals that there are more projects using `Node10` module resolution than `Node16`, `NodeNext`, and `Bundler` combined [36k using `Node10`](https://github.com/search?q=path%3A%2F%5Etsconfig.*.json%24%2F+language%3Ajson+%2F%22moduleResolution%22%3A+%22node%22%2F+OR+%2F%22moduleResolution%22%3A+%22node10%22%2F&type=code) versus [25k using other resolutions](https://github.com/search?q=path%3A%2F%5Etsconfig.*.json%24%2F+language%3Ajson+%2F%22moduleResolution%22%3A+%22node16%22%2F+OR+%2F%22moduleResolution%22%3A+%22nodenext%22%2F+OR+%2F%22moduleResolution%22%3A+%22bundler%22%2F&type=code).

Given that supporting it is relatively simple using the [`typesVersions`](https://github.com/andrewbranch/example-subpath-exports-ts-compat/tree/main/examples/node_modules/types-versions-wildcards) strategy, it seems like a low-effort way to ensure compatibility for a larger portion of users, at least in the short term.

Reply:
The better search would be to separate `node` from `node10`.
You'll find that there's just 298 hits for `node10` -- i.e. 298 projects that have INTENTIONALLY opted to use node v10 resolution.

The remaining 35.8k hits for `node` are more than likely people that setup their config a long time ago and haven't touched it because it looks like it's the right thing to use. People don't realise that it's an alias for `node10` and that it's the wrong thing to use.

And you'll find that almost all of these codebases could migrate to `Node16` or `Bundler` without any bugs.

We went through this already when we dropped support for `Node10` resolution. Many users complained because they thought we had broken types, but we educated them about their incorrect configs, they updated it and people improved their setups.

We do not want to add it back because it is incorrect and misrepresents the types and accessibility of our package.

Reply:
Fair enough, I'll remove it.